app/services/audit/checksum_validator.rb

```diff
@@ -0,0 +1,301 @@
+    def latest_moab_storage_object_version
+      @latest_moab_storage_object_version ||= moab_on_storage.version_list.last
+    end
+
+    # @return [Boolean] false if the moab exists, true otherwise
+    def moab_on_storage_absent?
+      !File.exist?(object_dir) || latest_moab_storage_object_version.nil?
+    end
+
+    def update_complete_moab_status(status)
+      status_handler.update_status(status)
+    end
+
+    # Moab on storage and complete moab versions match?
+    def versions_match?
+      moab_on_storage.current_version_id == complete_moab.version
+    end
+
+    def validate_versions
+      # set_status_as_seen_on_disk will update results and complete_moab
+      status_handler.set_status_as_seen_on_disk(found_expected_version: versions_match?, moab_on_storage_validator: moab_on_storage_validator,
+                                                caller_validates_checksums: true)
+
+      return if versions_match?
+      results.add_result(AuditResults::UNEXPECTED_VERSION,
+                         actual_version: moab_on_storage.current_version_id,
+                         db_obj_name: 'CompleteMoab',
+                         db_obj_version: complete_moab.version)
+    end
+
+    def validate_manifest_inventories
+      moab_on_storage.version_list.each { |moab_version| ManifestInventoryValidator.validate(moab_version: moab_version, checksum_validator: self) }
+    end
+
+    def validate_signature_catalog
+      SignatureCatalogValidator.validate(checksum_validator: self)
+    end
+
+    def moab_on_storage_validator
+      @moab_on_storage_validator ||= MoabOnStorage::Validator.new(moab: moab_on_storage, audit_results: results)
+    end
+
+    def status_handler
+      @status_handler ||= StatusHandler.new(audit_results: results, complete_moab: complete_moab)
+    end
+
+    def moab_on_storage
+      @moab_on_storage ||= MoabOnStorage.moab(storage_location: storage_location, druid: druid)
+    end
+
+    def results
+      @results ||= AuditResults.new(druid: druid, moab_storage_root: moab_storage_root, actual_version: moab_on_storage.current_version_id,
+                                    check_name: 'validate_checksums')
+    end
+
+    def persist_db_transaction!(clear_connections: false)
+      # This is to deal with db connection timeouts.
+      ActiveRecord::Base.clear_active_connections! if clear_connections
+
+      transaction_ok = ActiveRecordUtils.with_transaction_and_rescue(results) do
+        yield if block_given?
+        complete_moab.save!
+      end
+      results.remove_db_updated_results unless transaction_ok
+      AuditResultsReporter.report_results(audit_results: results, logger: logger)
+    end
+
+    def object_dir
+      @object_dir ||= MoabOnStorage.object_dir(storage_location: storage_location, druid: druid)
+    end
+
+    def logger
+      @logger ||= Logger.new($stdout).extend(ActiveSupport::Logger.broadcast(Logger.new(Rails.root.join('log', 'cv.log'))))
```

Comment:
Gets its own logger, instead of getting from ChecksumValidatorUtils for less entanglement.

Reply:
how about renaming the log to checksum_validator (and ensure log rotation and all that)

also make sure that change is propagated to any log rotation in qa, stage and prod environments

Reply:
or maybe making a separate ticket for that?

Reply:
@ndushay Any `*.log` files in the app `log/` directory should automatically be rotated. So we're clear to keep this name or change it w/o any need for further work or ticketing, AFAICT.